guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc

```diff
@@ -0,0 +1,23 @@
+[id="prerequisites-for-configuring-{project-context}-with-keycloak-authentication_{context}"]
+= Prerequisites for configuring {Project} with {Keycloak} authentication
+
+Before configuring {Project} with {Keycloak} external authentication, ensure that you meet the following requirements:
+
+* A working installation of {Keycloak} server that uses HTTPS instead of HTTP.
+* A {Keycloak} account with admin privileges.
+* A realm for {Project} user accounts created in {Keycloak}.
+* If the certificates or the CA are self-signed, ensure that they are added to the end-user certificate trust store.
+* Users imported or added to {Keycloak}.
++
+If you have an existing user database configured such as LDAP or Kerberos, you can import users from it by configuring user federation.
+ifndef::orcharhino[]
+For more information, see {RHDocsBaseURL}red_hat_build_of_keycloak/24.0/html/server_administration_guide/user-storage-federation[User Storage Federation] in the _Red{nbsp}Hat build of {Keycloak} Administration Guide_.
+endif::[]
++
+If you do not have an existing user database configured, you can manually create users in {Keycloak}.
+ifndef::orcharhino[]
+For more information, see {RHDocsBaseURL}red_hat_build_of_keycloak/24.0/html/server_administration_guide/assembly-managing-users_server_administration_guide#proc-creating-user_server_administration_guide[Creating New Users] in the _Red{nbsp}Hat build of {Keycloak} Administration Guide_.
+endif::[]
+* This documentation relies on a {Keycloak} system setup with `--http-relative-path=/auth` configured. If you want to use a different context path, you will need to make manual adjustments or need to manually setup the httpd configuration.
```

Comment:
If you want to help make sure this bullet point doesn't get overlooked, consider moving it to the user story introduction: assembly_configuring-quarkus-based....adoc. It doesn't seem like a prerequisite anyway -- a prerequisite is more of an an action to be taken _before_ you perform the configuration procedure; this, on the other hand, is more of a pre-existing condition, so better suited for the introduction. IMHO :)

Also, a few further suggestions regarding the wording:

```suggestion
* Configuring Quarkus-based {Keycloak} authentication requires the `--http-relative-path=/auth` context path to be configured in your {Keycloak} system setup.
If you want to use a different context path, make manual adjustments or set up the `httpd` service manually.
```

To explain why I'm recommending this:

* "This documentation" -- This can be confusing because "this documentation" can also actually apply to the whole guide, or everything on docs.theforeman.org... It's better to be specific about what part of documentation you're referring to: "Configuring Quarkus-based authentication"
* "you will need to" -- We tend to avoid future tense because simple present usually works just fine; also, imperative voice for actions to be take by the user is also shorter and easier to understand.
* "setup httpd configuration" -- 'setting up configuration' feels like using the same word twice

Reply:
> If you want to help make sure this bullet point doesn't get overlooked, consider moving it to the user story introduction: assembly_configuring-quarkus-based....adoc. It doesn't seem like a prerequisite anyway -- a prerequisite is more of an an action to be taken before you perform the configuration procedure; this, on the other hand, is more of a pre-existing condition, so better suited for the introduction. IMHO :)

Okay this would mean not directly into the assembly, but in the `con_configuring-quarkus-keycloak-authentication-for-project.adoc` because it applies to all 3 paths
The only thing to think about here then is, should we also directly move the certificate message up, because it's really the same case, if you don't have a Keycloak running on HTTPS it won't work.
Another option I'm thinking of is to make it the 2nd bullet point in the requirements, right after the HTTPS one?

> "setup httpd configuration" -- 'setting up configuration' feels like using the same word twice

The 1st 2, yep I will correct these, really might be the better way to write it :)
The 3rd, when thinking more about that, might even be better to write it even more exactly, because if the file-name of the config doesn't apply to the scheme of `${apache::confd_dir}/${keycloak_app_name}_oidc_keycloak_${keycloak_realm}.conf`, foreman-installer will just remove it on the next run.
Or is it best to keep it vague?

Reply:
> Okay this would mean not directly into the assembly, but in the `con_configuring-quarkus-keycloak-authentication-for-project.adoc` because it applies to all 3 paths The only thing to think about here then is, should we also directly move the certificate message up, because it's really the same case, if you don't have a Keycloak running on HTTPS it won't work. Another option I'm thinking of is to make it the 2nd bullet point in the requirements, right after the HTTPS one?

Ok, after staring at the list for a few more minutes, making it the 2nd bullet point does seem to make more sense :) A good idea.

> 
> > "setup httpd configuration" -- 'setting up configuration' feels like using the same word twice
> 
> The 1st 2, yep I will correct these, really might be the better way to write it :) The 3rd, when thinking more about that, might even be better to write it even more exactly, because if the file-name of the config doesn't apply to the scheme of `${apache::confd_dir}/${keycloak_app_name}_oidc_keycloak_${keycloak_realm}.conf`, foreman-installer will just remove it on the next run. Or is it best to keep it vague?

Agreed, sounds important enough to be mentioned.